### PR TITLE
feat: Add backward, forward and enter 360 image for navigation

### PR DESCRIPTION
**Pull request by nilscognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/BND3D-4639

## Description :pencil:

This add `canDo360Action `and `do360Action `to the public interface in `Cognite3DViewer`. These function should be used by react-components,

  `Forward`: When inside 360 image, forwards on the history list
  `Backward`: When inside 360 image, backwards on the history list
  `Enter`: : When outside 360 image,  go to current in the history list (where you exit from)
  `Exit`: When inside 360, exit.
  
 This should correspond to 4 buttons in the navigation toolbar. I will have this interface to make it simple and symmetric.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/api/src/utilities/Image360Action.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+/**
+ * 360 image action to be used by the CogniteViewer.canDo360Action and do360Action
+ */
+export enum Image360Action {
```
Comment (pramodcog) [inline viewer/packages/api/src/utilities/Image360Action.ts:9]: As it is related 360 Images, I would suggest to move the file to 360-images package(like viewer/packages/360-images/src/*)

File: viewer/packages/api/src/api-helpers/Image360ApiHelper.ts [hunk 2]
```diff
@@ -491,15 +492,43 @@ export class Image360ApiHelper<DataSourceT extends DataSourceType> {
...
+    }
+    if (action === Image360Action.Enter) {
+      if (insideImage) {
+        return false;
+      }
+    } else {
+      if (!insideImage) {
+        return false;
+      }
+    }
```
Comment (pramodcog) [inline viewer/packages/api/src/api-helpers/Image360ApiHelper.ts:508]: Can you refract `if` `else` to `switch` case, it is more appropriate for `switch` case

File: viewer/packages/api/src/utilities/Image360Action.ts (hunk 1 above)
Comment (nilscognite) [reply viewer/packages/api/src/utilities/Image360Action.ts:9]: I didn't understand the structure. Why is 360Helper class not in that package?

Comment (pramodcog) [reply viewer/packages/api/src/utilities/Image360Action.ts:9]: Well, as Image360Helper is a bridge between the `360-images` package and `Cognite3DViewer`, it is not in 360-images folder/package.
I would suggest to move all implementation of features on 360 images should be in 360-images like `Image360History` and `Image360Action`

Comment (nilscognite) [reply viewer/packages/api/src/utilities/Image360Action.ts:9]: Ok.

File: viewer/packages/api/src/api-helpers/Image360ApiHelper.ts (hunk 2 above)
Comment (nilscognite) [reply viewer/packages/api/src/api-helpers/Image360ApiHelper.ts:508]: Done

File: viewer/packages/360-images/src/Image360Action.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * 360 image action to be used by the CogniteViewer.canDo360Action and do360Action
+ * @beta
+ */
+export enum Image360Action {
+  Forward, // When inside 360 image, forwards on the history list
+  Backward, // When inside 360 image, backwards on the history list
+  Enter, // When outside 360 image, go to current in the history list (where you exit from)
+  Exit // When inside 360, exit.
+}
```
Comment (pramodcog) [inline viewer/packages/360-images/src/Image360Action.ts:14]:
```suggestion
export enum Image360Action {
  /**
  * When inside 360 image, forwards on the history list.
  */
  Forward,
  /**
  * When inside 360 image, backwards on the history list.
  */
  Backward,
  /**
  * When outside 360 image, go to current in the history list (where you exit from).
  */
  Enter,
  /**
  * When inside 360, exit.
  */
  Exit
}
```

Comment (pramodcog) [reply viewer/packages/360-images/src/Image360Action.ts:14]: This suggestion will removed undocumented tag in reveal.api.md file

Comment (nilscognite) [reply viewer/packages/360-images/src/Image360Action.ts]: Fixed

**Review by pramodcog — APPROVED** [review]
LGTM!

**Pull request merged**